pkg/resources/api_authentication_integration_with_authorization_code_grant.go

```diff
@@ -0,0 +1,523 @@
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		req.WithOauthClientAuthMethod(value)
+	}
+
+	if v := d.Get("oauth_refresh_token_validity").(int); v != -1 {
+		req.WithOauthRefreshTokenValidity(v)
+	}
+
+	if v := d.Get("oauth_grant").(string); v != "unknown" {
+		if v == "AUTHORIZATION_CODE" {
+			req.WithOauthGrantAuthorizationCode(true)
+		}
+	}
+
+	if v := d.Get("oauth_token_endpoint").(string); v != "unknown" {
+		req.WithOauthTokenEndpoint(v)
+	}
+	if v, ok := d.GetOk("oauth_allowed_scopes"); ok {
+		elems := expandStringList(v.(*schema.Set).List())
+		allowedScopes := make([]sdk.AllowedScope, len(elems))
+		for i := range elems {
+			allowedScopes[i] = sdk.AllowedScope{Scope: elems[i]}
+		}
+		req.WithOauthAllowedScopes(allowedScopes)
+	}
+
+	if err := client.SecurityIntegrations.CreateApiAuthenticationWithAuthorizationCodeGrantFlow(ctx, req); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(name)
+
+	return ReadContextApiAuthenticationIntegrationWithAuthorizationCodeGrant(false)(ctx, d, meta)
+}
+
+func ReadContextApiAuthenticationIntegrationWithAuthorizationCodeGrant(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+		integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to query security integration. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Security integration name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
+		properties, err := client.SecurityIntegrations.Describe(ctx, id)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if c := integration.Category; c != sdk.SecurityIntegrationCategory {
+			return diag.FromErr(fmt.Errorf("expected %v to be a %s integration, got %v", id, sdk.SecurityIntegrationCategory, c))
+		}
+		if err := d.Set("name", integration.Name); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("comment", integration.Comment); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("enabled", integration.Enabled); err != nil {
+			return diag.FromErr(err)
+		}
+		if withExternalChangesMarking {
+			if err = handleExternalChangesToObjectInShow(d,
+				showMapping{"comment", "comment", integration.Comment, integration.Comment, nil},
+				showMapping{"enabled", "enabled", integration.Enabled, integration.Enabled, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+
+			enabled, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "ENABLED" })
```

Comment:
why we handle enabled from two requests?

Reply:
Removed handling from describe, will add it to our docs.